### ISSUE TITLE
assigning_clones: unhelpful in test code

### ISSUE THREAD

**Issue by austinjones** [rust-lang/rust-clippy#12752]
### Summary

I ran into a case where assigning_clones is unhelpful, and most of our test modules are going to need `#[allow(clippy::assigning_clones)]`.

This occurs when a test struct is initialized with some default, and you need to assign a string to one field.  This inverts the order of assignee/assignment and makes tests much harder to read, and more complicated to write.

Perhaps this check should be disabled for test code?

### Reproducer

I tried this code:

```rust
fn main() {}

#[cfg(test)]
mod test {
    #[derive(Default)]
    struct Data {
        field: String,
    }

    fn test_data() -> Data {
        Data {
            field: "default_value".to_string(),
        }
    }

    #[test]
    fn test() {
        let mut data = test_data();
        data.field = "override_value".to_owned();
    }
}
```

I ran `cargo clippy --all-targets`, and I saw this happen:

```
    Checking assigning-clones-repro v0.1.0 (/home/austinjones/open-source/assigning-clones-repro)
warning: assigning the result of `ToOwned::to_owned()` may be inefficient
  --> src/main.rs:19:9
   |
19 |         data.field = "override_value".to_owned();
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use `clone_into()`: `"override_value".clone_into(&mut data.field)`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assigning_clones
   = note: `#[warn(clippy::assigning_clones)]` on by default

warning: `assigning-clones-repro` (bin "assigning-clones-repro" test) generated 1 warning (run `cargo clippy --fix --bin "assigning-clones-repro" --tests` to apply 1 suggestion)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.06s
```

I expected to see this happen:
```
    Checking assigning-clones-repro v0.1.0 (/home/austinjones/open-source/assigning-clones-repro)
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 0.04s
```

### Version

```text
rustc 1.78.0 (9b00956e5 2024-04-29)
binary: rustc
commit-hash: 9b00956e56009bab2aa15d7bff10916599e3d6d6
commit-date: 2024-04-29
host: x86_64-unknown-linux-gnu
release: 1.78.0
LLVM version: 18.1.2
```
